### PR TITLE
[ALZ-159] TryCatch blocks for each possible error

### PR DESCRIPTION
**Pull request by erichare**
This PR splits the possible error conditions with separate tryCatch blocks, attempting to give an informative error message in the case of different types of errors

### CODE REVIEW

File: R/mod-main.R [hunk 1]
```diff
@@ -94,39 +83,102 @@ mod_main_server <- function(input, output, session, syn) {
...
+        Sys.sleep(2)
+        waiter::waiter_update(
+          html = tagList(
+            img(src = "www/synapse_logo.png", height = "120px"),
+            h3("Submission Display error"),
+            span(
+              paste0("There was an error displaying submissions: ", err),
+              a("Synapse", href = "https://www.synapse.org/", target = "_blank"),
+              ", then refresh this page. If the problem persists, contact an administrator."
+            )
```
Comment (JessterB) [inline R/mod-main.R:140]: And here:
```
h3("Submission Display Error"),
span(
  paste0("There was an error displaying submission data for scoring: ", err,
  "\n\n Please refresh this page. If the problem persists, contact an administrator."
)
```

File: R/mod-main.R [hunk 2]
```diff
@@ -94,39 +83,102 @@ mod_main_server <- function(input, output, session, syn) {
...
+        waiter::waiter_update(
+          html = tagList(
+            img(src = "www/synapse_logo.png", height = "120px"),
+            h3("View Panel error"),
+            span(
+              paste0("There was an error viewing the panel: ", err),
+              a("Synapse", href = "https://www.synapse.org/", target = "_blank"),
+              ", then refresh this page. If the problem persists, contact an administrator."
+            )
+          )
```
Comment (JessterB) [inline R/mod-main.R:179]: Error type and message for `mod_view_all_section_server` trycatch block:
```
3("View All Submissions Display Error"),
span(
  paste0("There was an error displaying submission data: ", err),
  "\n\n Please refresh this page. If the problem persists, contact an administrator."
)

```

File: R/mod-main.R [hunk 3]
```diff
@@ -94,39 +83,102 @@ mod_main_server <- function(input, output, session, syn) {
...
+          mod_panel_section_server,
+          "panel_section",
+          synapse = synapse,
+          syn = syn,
+          user = user,
+          submissions = sub_data,
+          reviews_table = "syn22014561",
+          submissions_table = "syn22213241"
+        )
+
```
Comment (JessterB) [inline R/mod-main.R:158]: isolate the call to `mod_panel_section_server` above in it's own trycatch block

Comment (JessterB) [reply R/mod-main.R:158]: Error type and message for `mod_panel_section_server` trycatch block:

```
h3("Summarized Scores Display Error"),
span(
  paste0("There was an error displaying summarized scores data: ", err),
  "\n\n Please refresh this page. If the problem persists, contact an administrator."
)

```

File: R/mod-main.R [hunk 4]
```diff
@@ -94,39 +83,102 @@ mod_main_server <- function(input, output, session, syn) {
...
+      Sys.sleep(2)
+      waiter::waiter_update(
+        html = tagList(
+          img(src = "www/synapse_logo.png", height = "120px"),
+          h3("Submissions Retrieval error"),
+          span(
+            paste0("There was an error processing submissions: ", err),
+            a("Synapse", href = "https://www.synapse.org/", target = "_blank"),
+            ", then refresh this page. If the problem persists, contact an administrator."
+          )
```
Comment (JessterB) [inline R/mod-main.R:110]: Let's adjust the Error type and message here:
```
h3("Submission Loading Error"),
span(
  paste0("There was an error loading submission data: ", err,
  "\n\n Please refresh this page. If the problem persists, contact an administrator."
)
```

**Review by JessterB — CHANGES_REQUESTED** [review]

**Comment by JessterB** [comment]
ALZ-180 has some related error handling changes, we can discuss at our sync today

**Comment by erichare** [comment]
@JessterB ready for re-review

File: R/data-gather-clean.R [hunk 5]
```diff
@@ -73,11 +73,22 @@ process_submissions <- function(submissions, lookup_table, complete = TRUE) {
...
+  ## Log the submission id and section for each row
+  sub_log <- dplyr::select(
+    all_subs,
+    submission_id = .data$form_data_id,
+    section = .data$section
+  )
+  
+  # Log it
+  print(sub_log)
+
```
Comment (JessterB) [inline R/data-gather-clean.R:85]: I'm not sure that this logging will be useful, as we aren't processing a specific submission or section here?

I guess I was expecting something like:
* Log the form_data_id at the beginning of create_table_from_json_file (which is called per submission from process_submissions)
* Log the section at the beginning of create_section_table (which is called per section from create_table_from_json_file)

**Review by JessterB — CHANGES_REQUESTED** [review]
For discussion today

File: R/data-gather-clean.R (hunk 5 above)
Comment (erichare) [reply R/data-gather-clean.R]: Just updated this! I'm just using prints right now, I was going to look into if there's a commonly used logging framework, though i know that the prints will show up in the shinyapps logs at least.

**Review by JessterB — APPROVED** [review]

**Pull request merged**